zstacklib/zstacklib/utils/file_metadata_handler.py

```
@@ -0,0 +1,273 @@
+            except Exception as e:
+                logger.warn("failed to stat metadata file %s: %s", fpath, e)
+
+        logger.debug("scan_vm_metadata on %s: found %d entries", metadataDir, len(entries))
+        return entries
+
+    def _do_cleanup(self, metadataPath):
+        # delete .json
+        try:
+            if os.path.exists(metadataPath):
+                os.remove(metadataPath)
+        except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)